samcli/lib/providers/cfn_base_api_provider.py

```diff
@@ -165,16 +182,26 @@ def extract_cors_http(self, cors_prop):
             allow_headers = self._get_cors_prop_http(cors_prop, "AllowHeaders", list)
             if isinstance(allow_headers, list):
                 allow_headers = ",".join(allow_headers)
+
+            # Read AllowCredentials but only output the header with the case-sensitive value of true
+            # (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials)
+            allow_credentials = "true" if self._get_cors_prop_http(cors_prop, "AllowCredentials", bool) else None
```

Comment:
Would getting rid of the `if` work?

If the prop exists and it's `True`, `_get_cors_prop` will return `true`, otherwise `None`. Seems identical here.

Reply:
Due to the RFC requiring that Access-Control-Allow-Credentials be the case sensitive value `true` and a boolean python value being converted to the value `True` when output, this is required to ensure correct casing of the output

Reply:
Gotcha, just noticed this one call `_get_cors_prop_http`, which type checks and passes the value through.